File: 2023/challenge_03/challenge_03.py
import os
import sys

def read_file():
  with open(
    os.path.join('2023', 'challenge_03', 'challenge_03.txt'),
    encoding='UTF-8'
  ) as file:
    return file.read().splitlines()
  
def run():
  try:
    param = int(sys.argv[1])
  except:
    print('⭕ Error: no se proporciono el parametro')
    sys.exit(1)

  content = read_file()
  count = 0

  for info_password in content:
    info, password = info_password.split(':')
    limit, word = info.split(' ')
    minimum, maximum = limit.split('-')

    if password.count(word) not in range(int(minimum), int(maximum)):
      count += 1
      # sys.argv[1]: parametro al ejecutar el script
      # py 2023/challenge_03/challenge_03.py 13
      if count == param: return password

if __name__ == '__main__':
  print(run())

File: 2023/challenge_03/challenge_03.js
const fs = require('node:fs/promises');
const path = require('node:path');

async function read_file() {
  const text = await fs.readFile(
    path.join(__dirname, 'challenge_03.txt'),
    'utf8'
  );
  return text.split('\n');
}

async function run() {
  const arg = Number(process.argv[2] ?? 13);
  const passwords = await read_file();
  let count = 0;

  for (const pass of passwords) {
    const [policy, password] = pass.split(':');
    const [range, letter] = policy.split(' ');
    const [min, max] = range.split('-');
    const letterCount = password.split('').filter(l => l === letter).length;
    
    if (!(min <= letterCount && letterCount <= max)) {
      count++;
      if (count === arg) {
        return password;
      }
    }
  }

  return '❗ No se encontro la contraseña';
}

run().then(console.log);